satpy/readers/seviri_base.py

```diff
@@ -686,9 +697,9 @@ def get_gain_offset(self):
                 internal_gain = gsics_gain
                 internal_offset = gsics_offset
 
-        if self._calib_mode == 'MEIRINK':
-            if coefs['MEIRINK'] is not None:
-                internal_gain = get_meirink_slope(coefs['MEIRINK'], self._scan_time)
+        if "MEIRINK" in self._calib_mode and self._channel_name in ['VIS006', 'VIS008', 'IR_016']:
```

Comment:
Could you please move that into a separate method `should_apply_meirink(calib_mode, channel_name)` or something like that